sim/common/vanilla/item_effects.go

```diff
@@ -252,28 +254,63 @@ func init() {
 			},
 		})
 
-		fistOfShahramAuras := character.NewPartyAuraArray(func(unit *core.Unit) *core.Aura {
+		/*fistOfShahramAuras := character.NewPartyAuraArray(func(unit *core.Unit) *core.Aura {
 			return unit.GetOrRegisterAura(core.Aura{
 				ActionID: core.ActionID{SpellID: 16601},
 				Label:    "Fist of Shahram",
 				Duration: time.Second * 8,
+				MaxStacks: 5,
 				OnGain: func(aura *core.Aura, sim *core.Simulation) {
 					character.MultiplyAttackSpeed(sim, 1.3)
 				},
 				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
 					character.MultiplyAttackSpeed(sim, 1/1.3)
 				},
 			})
-		})
+		})*/
+		
+
+
 		fistOfShahram := character.GetOrRegisterSpell(core.SpellConfig{
 			ActionID:    core.ActionID{SpellID: 16601},
 			SpellSchool: core.SpellSchoolArcane,
 			DefenseType: core.DefenseTypeMagic,
 			ProcMask:    core.ProcMaskEmpty,
 			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-				for _, aura := range fistOfShahramAuras {
+				/*for _, aura := range fistOfShahramAuras {
 					aura.Activate(sim)
+				}*/
+				
+				counter := 0
+				
+				for counter < 10 {
+					fistOfShahramAura := character.GetOrRegisterAura(core.Aura{
+						ActionID: core.ActionID{SpellID: 16601},
+						//Label:    "Fist of Shahram",  
+						//Label: fmt.Sprintf("Fist of Shahram (%s)(%f)", spell.ActionID.String(), sim.RandomFloat(spell.ActionID.String())),
+						Label: fmt.Sprintf("Fist of Shahram (%d)", counter),
+						Duration: time.Second * 8,
+						OnGain: func(aura *core.Aura, sim *core.Simulation) {
+							//character.PseudoStats.MeleeSpeedMultiplier *= 1.3
+							character.MultiplyAttackSpeed(sim, 1.3)
+						},
+						OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+							//character.PseudoStats.MeleeSpeedMultiplier /= 1.3
+							character.MultiplyAttackSpeed(sim, 1/(1.3))
+						},
+					})
+					
+					if !fistOfShahramAura.IsActive() {
+					    fistOfShahramAura.Activate(sim)
+					    break
+					}
+					
+					counter += 1
+					
 				}
```

Comment:
I'm a bit confused about what this is doing? I know you said that multiple instances of the aura can be active at once, but because this uses `GetOrRegisterAura` it will always use a single aura I'm pretty sure?

Reply:
The label is different (counter gets added to string) so they are unique and all have separate expiration times. They all share the same row in timeline (it gets darker the more that overlap) It works but feel feel to verify it in the timeline when you pull the branch.

Reply:
ooh ooh ok I see now. I missed that the first time around